h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -1623,7 +1762,7 @@ private double[] ADMM_solve(Gram gram, double[] xy) {
         }
         xy = xy.clone();
         chol.solve(xy);
-      } else {
+      } else {  // ADMM solve is only used when there is l1 regularization (lasso).
         gram = gram.deep_clone();
         xy = xy.clone();
         GramSolver slvr = new GramSolver(gram.clone(), xy.clone(), _parms._intercept, _state.l2pen(), _state.l1pen(), _state.activeBC()._betaGiven, _state.activeBC()._rho, _state.activeBC()._betaLB, _state.activeBC()._betaUB);
```

Comment:
Again, I see you cloned the object before, so what is the reason to clone them again? 

```suggestion
        GramSolver slvr = new GramSolver(gram, xy, _parms._intercept, _state.l2pen(), _state.l1pen(), _state.activeBC()._betaGiven, _state.activeBC()._rho, _state.activeBC()._betaLB, _state.activeBC()._betaUB);
```

Reply:
This is not related to what I am doing.  However, this needs investigation.  In addition, benchmarking should be performed to showcase the reduction in training.  I opened an issue here to deal with this: https://github.com/h2oai/h2o-3/issues/16142